src/main/java/org/hisp/dhis/rules/functions/RuleFunctionAddDays.java

```diff
@@ -67,7 +68,7 @@ public Object evaluate( ExprContext ctx, CommonExpressionVisitor visitor )
     public Object getDescription( ExprContext ctx, CommonExpressionVisitor visitor )
     {
         castDate( visitor.visit( ctx.expr( 0 ) ) );
-        castDate( visitor.visit( ctx.expr( 1 ) ) );
+        castDouble( visitor.visit( ctx.expr( 1 ) ) );
```

Comment:
@jimgrace
All looks good to me. Adding the number of days to date should result in the date instead of a number. Isn't this behavior correct?

Reply:
Yes, addDays should return a date. The lines here are just validating the arguments. Arg 1, referenced as `visitor.visit( ctx.expr( 0 ) )` should be a date. Arg 2, referenced as `visitor.visit( ctx.expr( 1 ) )` should be a number.